### ISSUE TITLE
Patch for libheif appears to have changed

### ISSUE THREAD

**Issue by nealef** [lovell/sharp-libvips#129]
In `build/lin.sh` there is:

```
$CURL https://github.com/strukturag/libheif/pull/583/commits/7da30e57498b2b67434abd4767377ee7b3d93ee4.patch
```

However, that no longer appears to exist and may have been superseded by:

[80300f8c8b4edb4e214a94668eeb9b88cba95774](https://github.com/strukturag/libheif/pull/583/commits/80300f8c8b4edb4e214a94668eeb9b88cba95774)

Also, while I'm in the neighbourhood, what level of `rust/cargo` is required?

**Comment by lovell** [comment]
Yes, thank you, updated with commit https://github.com/lovell/sharp-libvips/commit/4037fa655cd6c59ed30aa8a96e70b7c1ac1ad77d

What's the context of your question about rust?

**Comment by kleisauke** [comment]
FWIW, I did something similar on the `meson` branch:
https://github.com/lovell/sharp-libvips/compare/main...kleisauke:meson
https://github.com/kleisauke/libvips-packaging/commit/16d2b3bd54f850998cb922d12e92f9a455e656fb
https://github.com/kleisauke/libvips-packaging/commit/8c6d326247366a25c30920e54b80eff64d86de19

**Comment by lovell** [comment]
Thanks Kleis, your patch to allow librsvg to be built without docs should probably be submitted upstream to librsvg itself as I can imagine others will find that useful.

**Comment by nealef** [comment]
> FWIW, I did something similar on the `meson` branch: [main...kleisauke:meson](https://github.com/lovell/sharp-libvips/compare/main...kleisauke:meson) [kleisauke@16d2b3b](https://github.com/kleisauke/libvips-packaging/commit/16d2b3bd54f850998cb922d12e92f9a455e656fb) [kleisauke@8c6d326](https://github.com/kleisauke/libvips-packaging/commit/8c6d326247366a25c30920e54b80eff64d86de19)

I also had to make these changes:
```
@@ -457,7 +457,7 @@ fi
 make install-strip

 mkdir ${DEPS}/cgif
-$CURL https://github.com/dloebl/cgif/archive/V${VERSION_CGIF}.tar.gz | tar xzC ${DEPS}/cgif --strip-components=1
+$CURL https://github.com/dloebl/cgif/archive/refs/tags/V${VERSION_CGIF}.tar.gz | tar xzC ${DEPS}/cgif --strip-components=1
 cd ${DEPS}/cgif
 CFLAGS="${CFLAGS} -O3" meson setup _build --default-library=static --buildtype=release --strip --prefix=${TARGET} ${MESON} \
   -Dtests=false
@@ -465,7 +465,7 @@ ninja -C _build
 ninja -C _build install

 mkdir ${DEPS}/vips
-$CURL https://github.com/libvips/libvips/releases/download/v${VERSION_VIPS}/vips-${VERSION_VIPS}.tar.gz | tar xzC ${DEPS}/vips --strip-components=1
+$CURL https://github.com/libvips/libvips/archive/refs/tags/v${VERSION_VIPS}.tar.gz | tar xzC ${DEPS}/vips --strip-components=1
 cd ${DEPS}/vips
 # Prevent exporting the g_param_spec_types symbol to avoid collisions with shared libraries
 printf "{\n\
```

**Comment by nealef** [comment]
> Yes, thank you, updated with commit [4037fa6](https://github.com/lovell/sharp-libvips/commit/4037fa655cd6c59ed30aa8a96e70b7c1ac1ad77d)
> 
> What's the context of your question about rust?

I was trying to add a new target `linux-s390x`. I initially tried building the 8.0.6 level of code (required by Kibana) but I was getting rust compilation errors. With 8.12.2 I don't using the latest rust available to me (1.59). 

Also, `meson` complains it needs python 3.7 or 3.8 now and 3.6 won't suffice to I had to install the SCLO rh-python38 packages.

**Comment by lovell** [comment]
Closing as prebuilt binaries for the `linux-s390x` target will be made available from sharp v0.33.0.
